zig.hook: rename from zigHook

figsoda opened the pull request:
This makes using specific versions of zig easier (without overrides)

## Description of changes

since `zigHook` was added very recently, I didn't add an alias

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

figsoda committed:
zig.hook: rename from zigHook

This makes using specific versions of zig easier (without overrides)

AndersonTorres commented:
@ofborg build colorstorm

booniepepper commented on pkgs/tools/misc/poop/default.nix:
```diff
@@ -2,7 +2,7 @@
 , fetchFromGitHub
 , fetchpatch
-, zigHook
+, zig
```
My experience so far is teaching me that there is not going to be a forwards-compatible Zig introduced any time soon

Maybe these should all be explicit versions so that we don't have to change N packages the next time Zig has a release and we want to change the default Zig version. Maybe to put it another way ... Maybe there's no value yet in having a "default version" of Zig

AndersonTorres replied:
Let's pick a common ground:
The Nix packages should set Zig version explicitly, **but** for the end user it is cool to fire a `nix shell .#pkgs zig`.

figsoda replied:
good idea, I pushed the change as a separate commit

figsoda committed:
treewide: use zig_0_11 instead of zig for zig packages

Zig is still under active development, and new releases often introduce
breaking changes. This makes updating the default version of zig easier.

Some packages did not receive this change because they could be using
the c compiler or linker of zig, which doesn't receive as much breaking
changes compared to e.g. the zig std library.

https://github.com/NixOS/nixpkgs/pull/248243#discussion_r1289401340

AndersonTorres commented:
And the best test ever:

@ofborg build poop

figsoda commented:
I wish I can react with :hankey: on github

AndersonTorres commented:
Well, :hankey: happens
